### PR TITLE
Documented robottelo decorators

### PR DESCRIPTION
**Pull request by renzon**
Close: #3674

stubbed, skip_if_bug_open and skip_if_os were documented

Created a new section "Features" on documentation to
describe features from a high level point of view

### CODE REVIEW

**Comment by renzon** [comment]
I am not sure about the section name "Features". But the main idea is to create a entire section where we can document how the framework works. I've create it because there was existing section in which this content would be a nice fit.

File: docs/features/decorators.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+This section explains Robottello decorators.
+
+.. contents::
+
+Modules
+-------
+
+Robottelo decorators are located under
+:doc:`decorator package </api/robottelo.decorators>`. Most of then are used to
```
Comment (sghai) [inline docs/features/decorators.rst]: `s/then/them`

File: docs/features/decorators.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+control if a test must be skipped or executed accordingly with specific
+configurations.
+
+stubbed
+-------
+
+``stubbed`` skips any test it decorates. A reason can be provided as parameter
+and its default is "Not Implemented". Most of times it is used to define all
+test cases regarding a feature. Example::
```
Comment (sghai) [inline docs/features/decorators.rst]: `Most of times it is used to define all manual(not automated) tests related to a feature` ?

File: docs/features/decorators.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    @stubbed('Some other reason than not implemented')
+    def test_positive_create_matcher_attribute_priority(self):
+        """Test will be implemented later"""
+
+skip_if_bug_open
+----------------
+
+``skip_if_bug_open`` skips test based on bug tracker status. Currently Bugzilla
+and Redmine are supported. The decorator receives two parameters. The first
+indicates the tracker and must be onf of 'bugzilla' or 'redmine'. The second
```
Comment (sghai) [inline docs/features/decorators.rst]: `s/onf/one`

File: docs/features/decorators.rst [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        """
+
+    @skip_if_bug_open('redmine', 23456789)
+    def test_positive_add_puppet_content(self):
+        """Attempt to associate puppet repos within a custom content
+        view directly
+        """
+
+Decorator communicates with bug tracker to know the status of respective bug
+id. If it is still open, test will be skipped. Once dev team fix and change bug
```
Comment (sghai) [inline docs/features/decorators.rst]: Once bz is fixed and moved to `ON_QA` or `Verified` state, test will automatically be run with next builds. ?

File: docs/features/decorators.rst [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+status, test will run on next builds.
+
+This is useful to allow QE team automating test cases even before a bug is
+fixed or a feature is implemented, while test will not be considered a build
+fail until the issue is resolved.
+
+skip_if_os
+----------
+
+``skip_if_os`` skips test based on Foreman/Sattelite host os version. It
```
Comment (sghai) [inline docs/features/decorators.rst]: `s/sattelite/Satellite`

File: docs/features/decorators.rst [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        """Create Custom ostree repository"""
+
+The first test will be skipped if host os is RHEL6.x.y, where x and y can be
+any number. If ``RHEL6.1`` was used as parameter it would skip for any
+RHEL6.1.z version and so on.
+
+Arbitrary number versions can be passed as parameters. On second test both RHEL
+5 and 6 families would be skipped.
+
+This decorator is used to avoid false failures when an feature is available
```
Comment (sghai) [inline docs/features/decorators.rst]: `when a feature is supported only on one os version` ?

**Comment by sghai** [comment]
ACK pending a few minor comments, otherwise Looks good.

**Comment by sghai** [comment]
ACK.. Looks good.

**Comment by rochacbruno** [comment]
ACK

**Pull request merged**